src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -91,6 +92,16 @@
 
     private final EventBus eventBus = new EventBus();
 
+    private DatabaseLocation location;
+
+    public BibDatabase() {
+        this(DatabaseLocation.LOCAL);
+    }
+
+    public BibDatabase(DatabaseLocation location) {
```

Comment:
The BibDatabase class just represents a collection of entries and has no knowledge of how or where it is loaded from (so no location). All this context has to be stored in the BibDatabaseContext class.